scripts/refresh-skyux-eslint-deprecations-summary.ts

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable complexity, max-depth */
+import { glob } from 'glob';
+import { readFile, writeFile } from 'node:fs/promises';
+import path from 'node:path';
+
+interface DeprecatedProperty {
+  reason?: string;
+}
+
+type DeprecatedProperties = Record<string, DeprecatedProperty>;
+
+interface DeprecatedDirective {
+  deprecated: boolean;
+  reason?: string;
+  properties?: DeprecatedProperties;
+}
+
+interface Deprecations {
+  components?: Record<string, DeprecatedDirective>;
+  directives?: Record<string, DeprecatedDirective>;
+}
+
+const DEPRECATIONS_FILE_PATH = path.normalize(
+  'libs/sdk/skyux-eslint/src/__deprecations.json',
+);
+
+/**
+ * These selectors are deprecated, but have a non-deprecated replacement. Since
+ * the selectors are the same, we won't be able to distinguish between the
+ * deprecated selector and the non-deprecated selector.
+ */
+const EXCLUDE_SELECTORS = ['sky-page'];
+
+/**
+ * Refreshes the __deprecations.json file used by the
+ * 'skyux-eslint-template/no-deprecated-directives' rule.
+ */
+async function refreshSkyuxEslintDeprecationsSummary(): Promise<void> {
+  const deprecations: Deprecations = {};
+
+  const files = await glob('dist/libs/components/**/documentation.json');
```

Comment:
This can be `*` instead of `**`.

Reply:
Is there a meaningful difference between the two? I thought `**` meant (crosses directory boundaries).

Reply:
It's probably ok to leave it. If `documentation.json` is always at the top level of packages, this would save some extra traversal, but the packages don't go that deep anyway.